Src/ILGPU/Backends/SPIRV/BinarySPIRVBuilder.tt

```diff
@@ -0,0 +1,131 @@
+<#@ template hostspecific="true" language="C#" #>
+<#@ output extension=".cs"#>
+<#@ include file="BuilderCommon.ttinclude"
+#><#@ assembly name="System.Xml" #>
+<#@ assembly name="System.Core"#>
+<#@ import namespace="System.Linq"#>
+<#
+    void CreateBody(int opCode, List<(SPIRVOperand operand, string name)> paramInfos)
+    {
+#> {
+<#
+        PushIndent(standardIndent);
+        if (paramInfos.Count != 0)
+        {
+#>var tempList = new List<uint>();
+<#
+        }
+
+        foreach (var (_, name) in paramInfos)
+        {
+#>tempList.AddRange(SPIRVBuilderUtils.ToUintList(<#= name #>));
+<#
+        }
+#><#// Add 1 to the word count for the first word (what we are creating here)#>
+ushort opCode = <#= opCode#>;
+<#
+        if (paramInfos.Count != 0)
+        {
+#>ushort wordCount = (ushort) (tempList.Count + 1);
+<#
+        }
+        else
+        {
+#>ushort wordCount = 1;
+<#
+        }
+#>
+uint combinedWord = SPIRVBuilderUtils.JoinOpCodeWordCount(opCode, wordCount);
+_instructions.Add(combinedWord);
+<#
+        if (paramInfos.Count != 0)
+        {
+#>_instructions.AddRange(tempList);
+<#
+        }
+#>
+<#
+        PopIndent();
+#>
+}
+
+<#
+    }
+#>
+using System;
+using System.Collections.Generic;
+
+#nullable enable
+#pragma warning disable 1591
+
+namespace ILGPU.Backends.SPIRV {
+
+<#
+    PushIndent(standardIndent);
+#>
+/// <summary>
+/// Defines utility methods to generate SPIRV operations
+/// </summary>
+[CLSCompliant(false)]
+public class BinarySPIRVBuilder : ISPIRVBuilder {
+
+    private List<uint> _instructions = new List<uint>();
+
+    public byte[] ToByteArray() {
```

Comment:
Performance question: should be change to interface to return a `ReadOnlySpan<byte>`?

Reply:
I'm not sure if it will change the amount of copies we have to do in that function, but it will probably help somewhere else. I'll change it.